molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java

```diff
@@ -1056,13 +1068,21 @@ private void updateInternal(String entityName, String untypedId, Map<String, Obj
 		restService.updateMappedByEntities(entity, existing);
 	}
 
+	@Transactional
 	private void createInternal(String entityName, Map<String, Object> entityMap, HttpServletResponse response)
 	{
-		EntityType meta = dataService.getEntityType(entityName);
+		EntityType entityType = dataService.getEntityType(entityName);
+		Entity entity = this.restService.toEntity(entityType, entityMap);
 
-		Entity entity = this.restService.toEntity(meta, entityMap);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```

Comment:
Metadata == data, the REST controller shouldn't make any entity specific exceptions.